src/java.base/share/classes/sun/security/util/SecurityProviderConstants.java

```diff
@@ -105,13 +108,33 @@ public static final int getDefDSASubprimeSize(int primeSize) {
         }
     }
 
+    public static final int getDefAESKeySize() {
+        int currVal = DEF_AES_KEY_SIZE.get();
+        if (currVal == -1) {
+            int v = 256; // default AES key size
+            try {
+                // adjust if crypto policy only allows a smaller value
+                int max = Cipher.getMaxAllowedKeyLength("AES");
+                if (v > max)  {
+                    v = max;
+                }
+            } catch (NoSuchAlgorithmException ne) {
+                // should never happen; ignore and use the default
+            }
+            DEF_AES_KEY_SIZE.compareAndSet(-1, v);
+            currVal = v;
+        }
+        return currVal;
+    }
+
     public static final int DEF_DSA_KEY_SIZE;
     public static final int DEF_RSA_KEY_SIZE;
     public static final int DEF_RSASSA_PSS_KEY_SIZE;
     public static final int DEF_DH_KEY_SIZE;
     public static final int DEF_EC_KEY_SIZE;
     public static final int DEF_ED_KEY_SIZE;
     public static final int DEF_XEC_KEY_SIZE;
+    private static final AtomicInteger DEF_AES_KEY_SIZE;
```

Comment:
See the previous comment, maybe it could be
`public static final int DEF_AES_KEY_SIZE.`